### DISCUSSION TITLE
Get a list of missing translations from nuxt i18n - v9

### DISCUSSION THREAD

**Discussion by pocketstore-io** [nuxt-modules/i18n#3233] [Q&A]
Iam currently looking for a way to find all missing translations.
So i can provide translations for that.

Iam using v9.

Tried to add onMissing callback to the config, but doesnt work
Was looking with Chatgpt for a solution but doesnt find one.

Would appreciate any help on this, maybe there is a config option for this.

**Comment by pocketstore-io** [comment]
I found a way to  hack Nuxt3 via Plugin. So there should be a solution in the Config.

My Plugin for Nuxt 3:
export default defineNuxtPlugin((nuxtApp) => {
    nuxtApp.$i18n.setMissingHandler((locale,message) => {
        console.log(locale);
        console.log(message);
    });
})

**Comment by userquin** [comment]
I guess you should run a script with your translation files: your approach will force you to visit all pages per locale.

If you're using json files you can check this module https://github.com/elk-zone/elk/blob/main/scripts/prepare-translation-status.ts

To run the script (via `tsx`): https://github.com/elk-zone/elk/blob/main/package.json#L29 (you can remove TS and run it using `node ./scripts/prepare-translation-status.mjs`

You can change it to load js/ts files.

The result can be found here: https://docs.elk.zone/guide/contributing#translation-status (missing and outdated keys)

**Reply by userquin** [reply]
Maybe it will be nice to have this utility here with a new option to allow generate the summary.

**Comment by pocketstore-io** [comment]
i just wrote a few lines.

Iam using PocketBase.io for the Translations so i:
- establish a connect to pocketbase instance
- update the rules for creation translation
- and wrote the translation with local, message and a text for missing translations to it
- when i now go over every page the missing translation will be created and i only need to add the proper translation for it

But it would be nice to add a flag for a missing translation stored in the i18n folder with key for the translation.
